src/main/java/seedu/address/model/room/ReservationDates.java

```diff
@@ -0,0 +1,142 @@
+ * However, the removal of a ReservationDate uses ReservationDate#equals(Object) so
+ * as to ensure that the ReservationDate with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see ReservationDate#isOverlapping(ReservationDate)
+ */
+public class ReservationDates implements Iterable<ReservationDate> {
+
+    private final ObservableList<ReservationDate> internalList = FXCollections.observableArrayList();
+
+    /**
+     * Returns true if the list isOverlapping an equivalent ReservationDate as the given argument.
+     */
+    public boolean isOverlapping(ReservationDate toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isOverlapping);
+    }
+
+    /**
+     * Adds a ReservationDate to the list.
+     * The ReservationDate must not already exist in the list.
+     */
+    public void add(ReservationDate toAdd) {
+        requireNonNull(toAdd);
+        if (isOverlapping(toAdd)) {
+            throw new OverlappingReservationDateException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the ReservationDate {@code target} in the list with {@code editedReservationDate}.
+     * {@code target} must exist in the list.
+     * The ReservationDate identity of {@code editedReservationDate} must not be the same as another existing
+     * ReservationDate in the list.
+     */
+    public void setReservationDate(ReservationDate target, ReservationDate editedReservationDate) {
+        requireAllNonNull(target, editedReservationDate);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new ReservationDateNotFoundException();
+        }
+        // If edited reservation date is the same as the original, do nothing.
+        if (target.equals(editedReservationDate)) {
+            return;
+        }
+
+        if (isOverlapping(editedReservationDate)) {
+            throw new OverlappingReservationDateException();
+        }
+
+        internalList.set(index, editedReservationDate);
+    }
+
+    /**
+     * Removes the equivalent ReservationDate from the list.
+     * The ReservationDate must exist in the list.
+     */
+    public void remove(ReservationDate toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new ReservationDateNotFoundException();
+        }
+    }
+
+    public void setReservationDates(ReservationDates replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code ReservationDates}.
+     * {@code ReservationDates} must not contain duplicate ReservationDates.
+     */
+    public void setReservationDates(List<ReservationDate> reservationDates) {
+        requireAllNonNull(reservationDates);
```

Comment:
This class was refactored from `UniquePersonList` and thus retains similar methods, including the two setter methods mentioned here. I will be leaving it in because 1) to maintain consistency at this stage, 2) we might need it in the future for testing/other use cases.